### DISCUSSION TITLE
Task annotations?

### DISCUSSION THREAD

**Discussion by mgiugliano** [todotxt/todo.txt-cli#434] [Ideas]
Coming from Task Warrior, I am used to the advantages of task annotations. These are currently not possible in the ```todo.txt``` specs, if I am not mistaken. I am of course aware of the ["notes" action](https://github.com/todotxt/todo.txt-cli/wiki/Todo.sh-Add-on-Directory#note-take-notes-related-to-tasks) but that involves one extra file per task. I am also following the discussion for including URI or at least not triggering unwanted consequences when having ```https://```, ```mailto:*```, etc.

Inspired by comments in programming languages, I wonder whether a simple solution for annotating a task would be to add ```#``` at the very end of the line and then interpreting all that follows as an annotation. 

I hope it helps or triggers a discussion and I do hope from learn from you about this beautiful plain-text minimalistic and elegant project.

**Comment by inkarkat** [comment]
> the ["notes" action](https://github.com/todotxt/todo.txt-cli/wiki/Todo.sh-Add-on-Directory#note-take-notes-related-to-tasks) involves one extra file per task

So what? Isn't that a pure implementation detail that you shouldn't be concerned of? The use case for most people around annotations is longer text and/or multi-line text (which would be in conflict with the line-based todo.txt format or at least a hassle to handle).
If all you ask for is the possibility to ignore some task text parts on some outputs while showing the full task text on demand, using a custom marker (like `#`) coupled with a _custom filter_ and `todo.sh append "# annotation here"` to add those.

**Reply by mgiugliano** [reply]
Apologies for having bothered you. Forget about my naive and stupid remark. I won't be concerned.

**Reply by inkarkat** [reply]
There are no stupid questions... I'm genuinely interested in the use cases around task annotations. (I've personally never used them so far; I occasionally link external files inside the task, but never felt a need to directly attach those files closer to the task.)
